src/modules/sub/crypto.js

```diff
@@ -1,5 +1,26 @@
-import { createHmac } from "crypto";
+import { createHmac, createCipheriv, createDecipheriv, scryptSync } from "crypto";
+
+const algorithm = "aes256"
+const keyLength = 32;
 
 export function sha256(str, salt) {
-    return createHmac("sha256", salt).update(str).digest("hex");
+    return createHmac("sha256", salt).update(str).digest("base64");
+}
+
+export function encryptStream(str, iv, secret) {
+    const buff = Buffer.from(JSON.stringify(str), "utf-8");
+
+    const key = scryptSync(Buffer.from(secret, "base64"), "salt", keyLength);
```

Comment:
why not just generate a 32byte secret in createStream() and use that directly

Reply:
you're right, then there's no need to pass the secret anywhere

Reply:
idk what to do about this i can't think of anything